File: sdk/node/src/resources/health.ts
import type { BaseWingman, RequestOptions } from "../client.js";

export abstract class APIResource {
  protected _client: BaseWingman;
  constructor(client: BaseWingman) {
    this._client = client;
  }
}

export interface HealthResponse {
  status: string;
  [key: string]: unknown;
}

export class Health extends APIResource {
  /** GET /api/health */
  async check(options?: RequestOptions): Promise<HealthResponse> {
    return this._client.get<HealthResponse>("/api/health", options);
  }
}

File: sdk/python/src/wingman/resources/health.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any, Dict

if TYPE_CHECKING:
    from ..client import Wingman


HealthResponse = Dict[str, Any]


class APIResource:
    def __init__(self, client: "Wingman") -> None:
        self._client = client


class Health(APIResource):
    def check(self) -> HealthResponse:
        """GET /api/health"""
        return self._client.get("/api/health")
